composites/SnippetPreview/components/Icon.js

```diff
@@ -0,0 +1,31 @@
+import React from "react";
+import PropTypes from "prop-types";
+import styled from "styled-components";
+
+/**
+ * Returns the Icon component.
+ *
+ * @param {object} props Component props.
+ *
+ * @returns {ReactElement} Icon component.
+ */
+export const Icon = ( props ) => {
+	const IconComponent = styled( props.icon )`
+		float: left;
+		width: ${ props => props.iconSize }px;
+		height: ${ props => props.iconSize }px;
+		fill: ${ props => props.iconColor };
+	`;
+
+	return <IconComponent aria-hidden="true" iconSize={ props.iconSize } iconColor={ props.iconColor } />;
+};
+
+Icon.propTypes = {
+	icon: PropTypes.func.isRequired,
+	iconColor: PropTypes.string.isRequired,
+	iconSize: PropTypes.number,
```

Comment:
Can we make iconSize a string? Right now, to actually use this component you have to use this syntax:
` icon={ edit } iconColor="#f00" iconSize={ 40 }`
and remember to pass color as string but size as number. Not the most intuitive and easy to remember syntax.